lib/ansible/modules/system/etc_hosts.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/python
+# encoding: utf-8
+
+# (c) 2016, Jiri Tyr <jiri.tyr@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+from ansible.module_utils._text import to_bytes
+import os
+import re
+import tempfile
+
+
+ANSIBLE_METADATA = {
+    'status': [
+        'preview'
+    ],
+    'supported_by': 'community',
+    'version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: etc_hosts
+author: Jiri Tyr (@jtyr)
+version_added: '2.3'
+short_description: Manages records in the /etc/hosts file
+description:
+  - Add, remove and modify records in the /etc/hosts file.
+options:
+  ip:
+    required: true
+    description:
+      - IP address.
+  hostname:
+    required: false
+    description:
+      - Canonical hostname of the IP address. Required if I(state=present).
+  alias:
+    required: false
+    default: null
+    description:
+      - Alias(es) of the IP address. Multiple aliases can be either a normal
+        YAML list or a string containing comma separated list.
+  path:
+    required: false
+    default: /etc/hosts
```

Comment:
I have changed the name of the module from `hosts` to `etc_hosts` as suggested by @abadger because the name `hosts` seems to be too generic. I'm open to suggestions if you can think of a better name than `hosts` or `etc_hosts`.